feat: returning HTMLResponse for rendered block in FastAPI

s0er3n opened the pull request:
I think its a bit more intutive if the rendered block also returns somekind of response. Currently calling `templates.TemplateResponse()`does not return a `Response` if we only want to have one block rendered. Instead it returns a string. This would be especially useful if you want to conditionally render a block or the full content in the same route. Currently the args `status_code`,`headers`, `media_type` and `background` are unused when only rendering a fragment which could lead to confusion. Whats your opinion? If you like the PR we shouldn't forget to update the readme :)

sponsfreixes commented:
I've been looking at FastAPI and starlette's code, and it makes sense that if when working with "vanilla" Jinja we deal with instances of `startlette.templating._TemplateResponse`, which are subclassing from `startlette.responses.Response` with `media_type = "text/html"`, we should be doing the same thing.

I noticed that `startlette.responses.HTMLResponse` is adding that media type, so I wonder why `_TemplateResponse` it's not inheriting from that. Maybe an oversight?

Anyway, back to this PR, yes I like this idea. I was already not a fan of having `Union[_TemplateResponse, str]` as return type. I wonder if we can simplify that to `Response?` (If starlette was subclassing from `HTMLResponse` we could just use that... maybe they will accept a PR?).

Hopefully this change does not break any existing usage?

sponsfreixes commented:
I went ahead and asked about it: https://github.com/encode/starlette/discussions/2272

sponsfreixes commented:
I submitted a PR and got approved and merged into Starlette master.

On our side, for now, to make it compliant with the current and previous versions, we could stick to `Response` as the return type, and in the future change it to `HTMLResponse` to be more specific. Thoughts?

s0er3n commented:
Thanks for doing the PR and asking Questions. I think using Response now and later HTMLResponse is the way to go.

s0er3n commented:
I am now returning a response and changed the documentation. Also I copied one of your tests and added `response_class=HTMLResponse` to the decorator to check if it's not breaking current usage and its passing :)
